framework/resource/retryresource/crud_resource_wrapper.go

```diff
@@ -0,0 +1,100 @@
+package retryresource
+
+import (
+	"context"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	originalframework "github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/internal"
+	"github.com/giantswarm/operatorkit/framework/resource/internal/framework"
+)
+
+// crudResourceWrapper is a specialized wrapper which wraps
+// *framework.CRUDResource.
+type crudResourceWrapper struct {
+	logger   micrologger.Logger
+	resource framework.Resource
+
+	backOff backoff.BackOff
+}
```

Comment:
There is specialized `crudResourceWrapper` which wraps Resources with underlying `*framework.CRUDResource`.

`resourceWrapper` handles all other cases.